File: lib/src/modules/profile/dtos/create_review_request.dart
class CreateReviewRequest {
  final int userId;
  final List<int> productIds;
  final Map<int, double> ratingMap;
  final Map<int, String> commentMap;

  CreateReviewRequest({
    required this.userId,
    required this.productIds,
    required this.ratingMap,
    required this.commentMap,
  });

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'productIds': productIds,
      'ratingMap': ratingMap.map((key, value) => MapEntry(key.toString(), value)),
      'commentMap': commentMap.map((key, value) => MapEntry(key.toString(), value)),
    };
  }
}

File: src/main/java/com/example/sneakerx/dtos/product/CreateReviewRequest.java
package com.example.sneakerx.dtos.product;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateReviewRequest {
    private Integer userId;
    private List<Integer> productIds;
    private Map<Integer, Double> ratingMap; // {productId: rating}
    private Map<Integer, String> commentMap;    // {productId: comment}
}
